Framework/src/CheckRunner.cxx

```diff
@@ -181,11 +180,17 @@ void CheckRunner::refreshConfig(InitContext& iCtx)
       // prepare the information we need
       auto infrastructureSpec = InfrastructureSpecReader::readInfrastructureSpec(updatedTree);
 
-      // TODO: use the config to reconfigure the check runner.
-      // TODO: in particular, reset mChecks and update it.
-      // TODO: Problem is that a lot of the logic is in the infrastructure generator.
-      // TODO: we should probably just preserve the checks list and update their state.
-      // TODO: also see if we should update the detector name
+      // Use the config to reconfigure the check runner.
+      // The configs for the checks we find in the config and in our map are updated.
+      // Topology changes are ignored: New checks are ignored. Removed checks are ignored.
+      for (const auto& checkSpec : infrastructureSpec.checks) {
+        // search if we have this check in this runner and replace it
+        if (mChecks.find(checkSpec.checkName) != mChecks.end()) {
+          auto checkConfig = Check::extractConfig(infrastructureSpec.common, checkSpec);
+          mChecks.emplace(checkConfig.name, checkConfig);
```

Comment:
From https://www.cplusplus.com/reference/map/map/emplace/ :
> The insertion only takes place if no other element in the container has a key equivalent to the one being emplaced (keys in a map container are unique).

I think that means you are not overwriting the configuration.

Reply:
You are totally correct. I guess it is a time to use the new `insert_or_assign`

Reply:
well no, it does an insert and does not construct the object. I will just erase the object if it is there.

Reply:
@knopers8 I let you make a final check